python/paddle/distributed/auto_parallel/api.py

```diff
@@ -90,9 +91,263 @@ def sharding_specs(self):
+        >>> labels = np.random.rand(BATCH_SIZE, CLASS_NUM).astype('float32')
+        >>> dataset = RandomDataset(images, labels, BATCH_SIZE)
+        >>> loader = paddle.io.DataLoader(dataset, batch_size=BATCH_SIZE)
+
+        >>> mesh = dist.ProcessMesh([0, 1], dim_names=["x"])
+        >>> layer = DemoNet(mesh)
+        >>> opt = paddle.optimizer.SGD(
+        ...     learning_rate=0.1, parameters=layer.parameters()
+        ... )
+        >>> loss_fn = nn.MSELoss()
+
+        >>> dist_model, dist_loader = dist.static_decorate(
+        ...     layer, loader, loss_fn, opt
+        ... )
+
+        >>> # training
+        >>> dist_model.train()
+        >>> for batch_id, (image, label) in enumerate(dist_loader()):
+        ...     # in train mode, executing the __call__ method will
+        ...     # update the parameters of the model and return the
+        ...     # loss
+        ...     loss = dist_model(image, label)
+
+        >>> # evaluation
+        >>> dist_model.eval()
+        >>> for batch_id, (image, label) in enumerate(dist_loader()):
+        ...     # in eval mode, executing the __call__ method will
+        ...     # return the loss
+        ...     loss = dist_model(image, label)
+        ...     print(loss)
+
+        >>> # prediction
+        >>> dist_model.predict()
+        >>> for batch_id, (image, label) in enumerate(dist_loader()):
+        ...     # in predict mode, executing the __call__ method will
+        ...     # return a dict that contains the outputs of the model,
+        ...     # where the value of "out0" is the first output.
+        ...     outs = dist_model(image)
+        ...     print(outs['out0'])
+
+        >>> # This case need to be excuted in multi-card environment
+        >>> # export CUDA_VISIBLE_DEVICES=0,1
+        >>> # python -m paddle.distributed.launch {test_case}.py
+    """
+    dist_model = DistModel(layer, loss, optimizer, strategy)
+
+    # convert dygraph model to static model
+    batch_size = loader.batch_sampler.batch_size
+    inputs_spec, labels_spec = dist_model._engine._prepare_data_spec(
+        loader.dataset, None, batch_size
+    )
+
+    if optimizer is not None and loss is not None:
+        # get the static graph in train mode
+        dist_model._engine.prepare(
+            inputs_spec, labels_spec, mode="train", init_parameters=False
+        )
+    if loss is not None:
+        # get the static graph in eval mode
+        dist_model._engine.prepare(
+            inputs_spec, labels_spec, mode="eval", init_parameters=False
+        )
+    # get the static graph in predict mode
+    dist_model._engine.prepare(
+        inputs_spec, None, mode="predict", init_parameters=False
+    )
+
+    # get DistributedDataLoader for static mode auto-parallelism
+    batch_size = dist_model._engine._validate_batch_size(batch_size)
+    dist_loader = dist_model._engine._prepare_dataloader(
+        loader.dataset, return_list=True, batch_size=batch_size
+    )
```

Comment:
Better move these lines into `__init__ of DistModel`

Reply:
Done.